Userland/Libraries/LibGUI/ComboBox.cpp

```diff
@@ -85,6 +85,7 @@ ComboBox::ComboBox()
     };
 
     m_list_window = add<Window>(window());
+    m_list_window->set_window_type(WindowType::Menu);
```

Comment:
This is not correct. The problem is that tool windows are drawn at the wrong level, and we shouldn't hack ComboBox to force itself above by using an unrelated window type.

Reply:
Yeah, the thing is that by default it's using `WindowType::Normal` and `WindowType::ToolWindow` z-order is higher than it. What do you think should be done instead?

Reply:
I think we should fix WindowServer to do the z-order correctly :)

Reply:
> I think we should fix WindowServer to do the z-order correctly :)

I have changes in the works that address this problem specifically. Also in the context of full screen windows. it removes the `WindowType::ToolWindow` entirely and makes it merely a styling option, and the changes aim to properly preserve z-order for these things. It's not as straight forward as I hoped, which is why I haven't pushed my changes yet ;) But yes, changing this to the menu type isn't quite right...